### PR TITLE
BIC-195 # homeInteraction setting no longer honoured

### PR DESCRIPTION
**Pull request by GeetaSajwan**
Home interaction is honoured everytime tries to navigate to home.

### CODE REVIEW

File: tests/support/index.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
             BIC: {
               siteVars: {
                 answerSpaceId:  20403,
-                answerSpace: 'Integration',
+                answerSpace: 'geeta',
```
Comment (jokeyrhyme) [inline tests/support/index.html]: Did you mean to commit this change?

File: tests/support/index.html [hunk 2]
```diff
@@ -37,6 +37,6 @@
         <div data-role="page" id="temp">Loading, please wait.</div>
         <script src="//d1c6dfkb81l78v.cloudfront.net/blink/require/6/require.min.js"></script>
         <!--<script src="//d1c6dfkb81l78v.cloudfront.net/blink/bic/3/1411018236186/bic.js"></script>-->
-        <script src="/integration/bic.js"></script>
+        <script src="/geeta/bic.js"></script>
```
Comment (jokeyrhyme) [inline tests/support/index.html]: Did you mean to commit this change?

**Comment by jokeyrhyme** [comment]
Hmmm, I don't think it's a good idea to be passing the `resolve()` and `reject()` functions from inside a Promise as arguments to other functions. However, I can see that this file had this weird pattern long before this PR, so I can appreciate that your additions are just being consistent. :)

It makes me want to file a separate PR just to get rid of that weird pattern...

File: src/bic/model/interaction.js [hunk 3]
```diff
@@ -244,11 +243,29 @@ The argument change event.
 
               resolve(model);
             } else {
-              model.defaultView(app.interactions.models, app.siteVars);
-              resolve(model);
+              if (app.has('homeScreen') && app.get('homeScreen') !== false && app.has('homeInteraction')) {
```
Comment (jokeyrhyme) [inline src/bic/model/interaction.js]: I see this exact condition pop up in 3 different `if` statements. Maybe there should be a new shared `app#hasHomeInteraction()` method that gets used instead? That'll help with future maintenance, as we won't have to update 3 different `if` statements later.

Comment (GeetaSajwan) [inline src/bic/model/interaction.js]: sure, I will change that.

**Comment by simonmarklar** [comment]
lgtm :+1:

**Comment by jokeyrhyme** [comment]
@GeetaSajwan has this had a UX review?

**Comment by GeetaSajwan** [comment]
Not yet...

**Comment by GeetaSajwan** [comment]
i will ask everybody tomorrow morning.

**Pull request merged**